File: SBiswas/Problems/Milestone1/Problem2.cpp
#include <iostream>

class Node {
    
    public:
    int data;
    Node* left;
    Node* right;
    
    Node(int val)
    {
        data = val;
        left = nullptr;
        right = nullptr;
    }
    
};

Node* insert(int val)
{
    Node* temp = new Node(val);
    return temp;
}

void preorder(Node* root)
{
    if(root == nullptr)
    {
        return;
    }
    
    std::cout<<root->data<<" - ";
    preorder(root->left);
    preorder(root->right);
}

void postorder(Node* root)
{
    if(root == nullptr)
    {
        return;
    }
    
    postorder(root->left);
    postorder(root->right);
    std::cout<<root->data<<" - ";
}

int main()
{
    Node* my_tree = new Node(1);
    
    // insert(my_tree,1);
    my_tree->left = insert(2);
    my_tree->right = insert(3);
    
    my_tree->left->left = insert(4);
    my_tree->left->right = insert(5);
    
    my_tree->right->left = insert(6);
    my_tree->right->right = insert(7);
    
    std::cout<<"Preorder Traversal: \n";
    preorder(my_tree);
    std::cout<<'\n';
    std::cout<<"Postorder Traversal: \n";
    postorder(my_tree);
    std::cout<<'\n';
    
    return 0;
}

File: Hritwik/Problems/Milestone1/Problem2.py
class Node:
    """
    A class to represent a node in a binary search tree.
    
    Attributes:
    ----------
    value : int
        The value stored in the node.
    left : Node or None
        The left child of the node.
    right : Node or None
        The right child of the node.
    """
    def __init__(self, value: int):
        self.value=value
        self.left=None
        self.right=None

# Create a binary tree to traverse through it in an inorder traversal manner
#
#         Root   1
#              /    \
#             2      3  
#            / \    /  \
#           4   5  6    7
#          /
#         8
#
root=Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)
root.right.left = Node(6)
root.right.right = Node(7)
root.left.left.left= Node(8)

# Inorder traversal of the above binary tree

class Solution:

    def __init__(self):
        # Initializes Solution object
        pass
    
    def preorderTraversal(self, root: None):
        self.response=[]
    
        def traverse(node):
            if node is None:         
                return None  
            self.response.append(node.value)          
            traverse(node.left)
            traverse(node.right)

        traverse(root)
        return self.response

    def postorderTraversal(self, root: None):
        self.response=[]
    
        def traverse(node):
            if node is None:         
                return None       
            traverse(node.left)
            traverse(node.right)
            self.response.append(node.value)     

        traverse(root)
        return self.response

    

solution = Solution()

preorder_traversed_list = solution.preorderTraversal(root)
print(f'preorder traversed list : {preorder_traversed_list}')

postorder_traversed_list = solution.postorderTraversal(root)
print(f'postorder traversed list : {postorder_traversed_list}')






